source.go

```diff
@@ -419,20 +420,30 @@ func (s Source) Doc(name string) (io.Reader, error) {
 	return b, nil
 }
 
-func patchSource(worker, sourceState llb.State, sourceToState map[string]llb.State, patchNames []PatchSpec, opts ...llb.ConstraintsOpt) llb.State {
-	for _, p := range patchNames {
-		patchState := sourceToState[p.Source]
-		// on each iteration, mount source state to /src to run `patch`, and
-		// set the state under /src to be the source state for the next iteration
-		sourceState = worker.Run(
-			llb.AddMount("/patch", patchState, llb.Readonly, llb.SourcePath(p.Source)),
-			llb.Dir("src"),
-			shArgs(fmt.Sprintf("patch -p%d < /patch", *p.Strip)),
-			WithConstraints(opts...),
-		).AddMount("/src", sourceState)
-	}
+type patchFunc func(worker, sourceState llb.State, sourceToState map[string]llb.State, patchNames []PatchSpec, opts ...llb.ConstraintsOpt) llb.State
```

Comment:
Not sure if we need the nested closure here? Maybe we could just add an extra `Source` or even `isDir` param to the original function?

Reply:
done